Doc/library/stdtypes.rst

```diff
@@ -4743,6 +4743,111 @@ define these methods must provide them as a normal Python accessible method.
 Compared to the overhead of setting up the runtime context, the overhead of a
 single class dictionary lookup is negligible.
 
+.. _types-union:
+
+Union Type
+====================
+
+.. index::
+   object: Union
+   pair: union; type
+
+A union object holds the value of the ``|`` (bitwise or) operation on
+multiple :ref:`type objects<bltin-type-objects>`. This enables cleaner type
+hinting syntax compared to :data:`typing.Union`.
+
+.. describe:: X | Y | ...
+
+   Defines a union object which holds types *X*, *Y*, and so forth. X | Y
+   means either X or Y. It is syntactically equivalent to
+   ``typing.Union[X, Y, ...]``.
+   Example::
+
+      def square(number: int | float) -> int | float:
+          return number ** 2
+
+.. describe:: union_object == other
+
+   Union objects can be tested for equality with other union objects. Details:
+
+   * Unions of unions are flattened, e.g.::
+
+       (int | str) | float == int | str | float
+
+   * Redundant arguments are skipped, e.g.::
+
+       int | str | int == int | str
+
+   * When comparing unions, the argument order is ignored, e.g.::
+
+      int | str == str | int
+
+   * Compatible with :data:`typing.Union`::
+
+      int | str == typing.Union[int, str]
+
+   * Optional values are equivalent to :data:`typing.Optional`::
+
+      str | None == typing.Optional[str]
+
+.. describe:: isinstance(obj, union_object)
+
+   Calls to :func:`isinstance` are also supported with a Union object::
+
+      >>> isinstance("", int | str)
+      True
+
+   Union objects containing parametrized generics cannot be used::
```

Comment:
Unless I've missed out something, the only documentation for parameterized generics I can find is in PEP585. I think it might not be documented yet (see: https://bugs.python.org/issue40814, https://bugs.python.org/issue41805).
 
I'm open to making a separate PR in the future for that (assuming the person working on typing.GenericAlias isn't anymore), but I'd like to wait for this PR to be resolved first to get an understanding of how I can write better documentation.

For now, I'll link it to typing.Generic

Reply:
Actually, I think that linking to PEP585 would be a better temporary solution, what do you think?

Reply:
Ah, that makes sense. Can you add a comment to the doc reminding future generations to update the link?